diff --git a/src/cmp.c b/src/cmp.c
index 26067d9..c15d1d7 100644
--- a/src/cmp.c
+++ b/src/cmp.c
@@ -312,6 +312,13 @@ main (int argc, char **argv)
           stat_buf[f].st_blksize = 8 * 1024;
 #endif
         }
+
+#ifdef __MVS__
+      if (S_ISFIFO(stat_buf[f].st_mode) && stat_buf[f].st_blksize < 0) 
+        {
+          stat_buf[f].st_blksize = 8 * 1024;
+        }
+#endif
       else
 	stat_buf[f].st_size = stat_size (&stat_buf[f]);
     }
