diff --git a/lib/stdlib.in.h b/lib/stdlib.in.h
index bd82086..72f521e 100644
--- a/lib/stdlib.in.h
+++ b/lib/stdlib.in.h
@@ -233,6 +233,20 @@ _GL_INLINE_HEADER_BEGIN
 #  endif
 _GL_FUNCDECL_RPL (_Exit, _Noreturn void, (int status), );
 _GL_CXXALIAS_RPL (_Exit, void, (int status));
+#if defined(__MVS__)
+  #ifndef __EDC_LE
+    #define __EDC_LE 0x10000000
+  #endif
+  #if __TARGET_LIB__ >= __EDC_LE
+    #ifdef __SUSV3
+      #ifdef __NATIVE_ASCII_F
+        #pragma map(unsetenv,"\174\174A00471")
+      #else
+        #pragma map(unsetenv,"\174\174UNSTNV")
+      #endif
+    #endif
+  #endif
+#endif
 # else
 #  if !@HAVE__EXIT@
 _GL_FUNCDECL_SYS (_Exit, _Noreturn void, (int status), );
