diff --git a/lib/stdlib.in.h b/lib/stdlib.in.h
index 652062d..c728caf 100644
--- a/lib/stdlib.in.h
+++ b/lib/stdlib.in.h
@@ -1401,6 +1401,20 @@ _GL_WARN_ON_USE (unlockpt, "unlockpt is not portable - "
 #  endif
 _GL_FUNCDECL_RPL (unsetenv, int, (const char *name) _GL_ARG_NONNULL ((1)));
 _GL_CXXALIAS_RPL (unsetenv, int, (const char *name));
+#if defined(__MVS__)
+  #ifndef __EDC_LE
+    #define __EDC_LE 0x10000000
+  #endif
+  #if __TARGET_LIB__ >= __EDC_LE
+    #ifdef __SUSV3
+      #ifdef __NATIVE_ASCII_F
+        #pragma map(unsetenv,"\174\174A00471")
+      #else
+        #pragma map(unsetenv,"\174\174UNSTNV")
+      #endif
+    #endif
+  #endif
+#endif
 # else
 #  if !@HAVE_DECL_UNSETENV@
 _GL_FUNCDECL_SYS (unsetenv, int, (const char *name) _GL_ARG_NONNULL ((1)));
