diff --git a/build-aux/config.guess b/build-aux/config.guess
index e81d3ae..41c08e8 100755
--- a/build-aux/config.guess
+++ b/build-aux/config.guess
@@ -379,7 +379,7 @@ case $UNAME_MACHINE:$UNAME_SYSTEM:$UNAME_RELEASE:$UNAME_VERSION in
 	GUESS=$UNAME_MACHINE-unknown-morphos
 	;;
     *:OS/390:*:*)
-	GUESS=i370-ibm-openedition
+	GUESS=s390x-ibm-openedition
 	;;
     *:z/VM:*:*)
 	GUESS=s390-ibm-zvmoe
