diff --git a/lib/intprops.h b/lib/intprops.h
index 9d10028..c878c3f 100644
--- a/lib/intprops.h
+++ b/lib/intprops.h
@@ -431,7 +431,7 @@
 /* Store the low-order bits of A <op> B into *R, where OP specifies
    the operation and OVERFLOW the overflow predicate.  Return 1 if the
    result overflows.  See above for restrictions.  */
-#if 201112 <= __STDC_VERSION__ && !_GL__GENERIC_BOGUS
+#if 201112 <= __STDC_VERSION__ && !_GL__GENERIC_BOGUS && !defined(__MVS__)
 # define _GL_INT_OP_WRAPV(a, b, r, op, overflow) \
    (_Generic \
     (*(r), \
